internal/tests/testrepository/header_testrepository.go

```diff
@@ -193,6 +194,51 @@ func (r *HeaderTestRepository) GetChainBetweenTwoHashes(low string, high string)
 	return headers, nil
 }
 
+// GetMerkleRoots returns ExclusiveStartKey pagination of batchSize size with merkle roots from lastEvaluatedKey which
+// is the last height of the block that a client has processed
+func (r *HeaderTestRepository) GetMerkleRoots(batchSize int, lastEvaluatedKey int) (*domains.MerkleRootsESKPagedResponse, error) {
+	merkleroots := make([]*domains.BlockHeader, 0)
+	orderByField := "BlockHeight"
+	sortDirection := "ASC"
+	// order headers by height ASC
+	sort.Slice(*r.db, func(i, j int) bool {
+		return (*r.db)[i].Height < (*r.db)[j].Height
+	})
```

Comment:
Just for the record (general comment) - storing a pointer to a slice is in almost every cases pointless, because slices are already "referential types" (can be `nil`).

But I'm aware that it was already implemented that way and you had to deal with it. 
That's why you had to use `*r.db` (with a "star" 🙈)